FinnUI/Sources/Components/ChatAvailability/ChatAvailabilityView.swift

```diff
@@ -0,0 +1,155 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import FinniversKit
+
+public class ChatAvailabilityView: UIView {
+    public enum Status: CaseIterable {
+        case loading
+        case online
+        case offline
+        case unknown
+    }
+
+    // MARK: - Private properties
+
+    private lazy var statusView = StatusView(withAutoLayout: true)
+
+    private lazy var stackView: UIStackView = {
+        let stackView = UIStackView(withAutoLayout: true)
+        stackView.axis = .vertical
+        stackView.spacing = .spacingS
+        return stackView
+    }()
+
+    private lazy var chatNowButton: Button = {
+        let button = Button(style: .callToAction, size: .normal, withAutoLayout: true)
+        button.addTarget(self, action: #selector(handleButtonTap), for: .touchUpInside)
+        button.titleEdgeInsets = UIEdgeInsets(leading: .spacingS)
+        button.imageEdgeInsets = UIEdgeInsets(top: .spacingXS, leading: -.spacingS)
+        let image = UIImage(named: .videoChat)
+        button.setImage(image.withRenderingMode(.alwaysTemplate))
+        button.imageView?.tintColor = .milk
+        return button
+    }()
+
+    // MARK: - Init
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    required init?(coder: NSCoder) { fatalError() }
+
+    // MARK: - Setup
+
+    private func setup() {
+        addSubview(stackView)
+        stackView.addArrangedSubview(chatNowButton)
+        stackView.addArrangedSubview(statusView)
+        stackView.fillInSuperview()
+    }
+
+    // MARK: - Public methods
+
+    public func configure(buttonTitle: String?) {
+        chatNowButton.setTitle(buttonTitle, for: .normal)
+    }
+
+    public func configure(status: Status, statusTitle: String? = nil) {
+        switch status {
+        case .online, .unknown:
+            chatNowButton.isEnabled = true
+        case .loading, .offline:
+            chatNowButton.isEnabled = false
+        }
+
+        statusView.configure(status: status, statusTitle: statusTitle)
+    }
+
+    // MARK: - Private methods
+
+    @objc private func handleButtonTap() {
+
```

Comment:
is there something missing here? a delegate/callback call, etc

Reply:
Hehe, forgot it during the all hands meeting😄 Added it a couple of minutes ago, though💪